src/fileManagement.js

```diff
@@ -318,4 +318,35 @@ files.prototype.copy = function (source, target) {
   })
 }
 
+/**
+ * Mark a remote file or directory as favorite
+ * @param   {string} path       path of file/folder
+ * @param   {boolean} value     Star or un-star the file/folder
+ * @returns {Promise.<status>}  boolean: whether the operation was successful
+ * @returns {Promise.<error>}   string: error message, if any.
+ */
+files.prototype.star = function (path, value) {
```

Comment:
we usually call it "favorite" everywhere in the core code, never "star". it's only the icon that is a star.
might be better to keep it consistent to avoid future confusion